docs/defaulting.md

```diff
@@ -0,0 +1,249 @@
+# Ahead of Time libvirt Defaulting on the Cluster/API Level
+
+This Proposal fills a gap between KubeVirt as Runtime and Higher level
+management, like proposed in
+[docs/config-and-instances.md](docs/config-and-instances.md). It lays out a
+design which overcomes the necessity for asynchronous specification updates,
+which has it's origin in the fact that a lot of the defaulting is done on the
+host by libvirt, so after scheduling.
+
+With the proposal here, which will provide a synchronous way of fully
+populating a VM specification, we come much closer to the meaning of a Pod
+specification in Kubernetes.
+
+Note that a fully populated VM specification in the context of this proposal, is
+a specification which has filled out all fields, which influence how the
+emulated machine looks like for the Guest OS. For instance libvirt network
+configurations may differ on a host to host basis, e.g. because on one host a
+path to an Ethernet device is `/dev/eth0` and on another one `/dev/eth1`,
+although in both cases we connect to the same network. Such backend
+configuration specific fields are not of interest for this proposal.
+
+The goals of this proposal are:
+
+ * Synchronous defaulting of all VM specification related fields, which
+   influence how a Guest OS sees a VM.
+ * As a consequence, enforcing the meaning of the VM specification as
+   specification. We then don't have to change the specification after it was
+   provided, or force users to extract additional data from the status field,
+   which they would have to put into the VM specification in subsequent runs,
+   to get exactly the same VM.
+ * Give us more control over what we want to expose in `status`, instead of
+   blindly copying every non-sensitive domain information into the status,
+   which host-only libvirt defaulting would force us to do.
+
+## Current situation
+
+libvirt allows you to post very minimal Domain specifications. It will then
+fill out a lot of device details for you. On cluster level this can lead to
+difficulties, since depending on the fact if the VM has ever run before, you
+will have to operate on different specs, since you have to incorporate VM
+specification changes once a VM was started the first time. Furhter, getting
```

Comment:
What would be sufficient for you? What I mean, is that you have to either wait asynchronously for additional spec updates from the cluster, which you need to save for later re-runs (either pull the additional data out from a status, or from an updated spec). So you have a VM Spec before the VM is started, one because of the POST, and one because of the first run, the same for PUTs on hotplug. That are the different ones I mean. Is this more understandable?